### DISCUSSION TITLE
Kage: Image limit

### DISCUSSION THREAD

**Discussion by pedro-tiple** [hajimehoshi/ebiten#2882] [Q&A]
Hello, I'm encoding data into images and using shaders to process it. So far things work mostly alright but now I've hit a case where I need more than four inputs, and it maxes out at four.

I've tried to raise the limit [here](https://github.com/hajimehoshi/ebiten/blob/fdf36026aee97e674f23c219f7cfc2a544b13f51/image.go#L546) and [here](https://github.com/hajimehoshi/ebiten/blob/fdf36026aee97e674f23c219f7cfc2a544b13f51/internal/graphics/vertex.go#L18), but that leads to a black screen with no errors.

From a quick search it looks like WebGL, Metal, DirectX, and OpenGL all allow 16 textures or more (depending on platform), is this limitation of four in the engine self imposed or is there some actual external limit?

Some pointers on what others changes would be required to increase this limit would be appreciated.
I'm fine with doing a local fork for personal use only, in case this is something that isn't desired for the whole engine.

Thanks!

**Comment by hajimehoshi** [comment]
> is this limitation of four in the engine self imposed or is there some actual external limit?

I determined the maximum as an arbitrary number. Please file a request if you want to increase this number.

**Reply by pedro-tiple** [reply]
Thanks for the quick reply.
Happy to file a request, however this is blocking some work I'd like to keep moving forward with, even if that implies making the changes myself locally as to not have to wait for you to have availability for it.

> Some pointers on what others changes would be required to increase this limit would be appreciated.

Doubling down on this one, to avoid having to lose more time than necessary trying to figure out what needs to change.

**Reply by hajimehoshi** [reply]
Updating `graphics.ShaderImageCount` from 4 to something should work. If not, there is something I am missing.
